### PR TITLE
fix killing of server process and its children, for travis-ci

### PR DESCRIPTION
**Pull request by yanivefraim**
### 🔦 Summary
On Linux, child processes of child processes will not be terminated when attempting to kill their parent (see [here](https://nodejs.org/api/child_process.html#child_process_subprocess_kill_signal))

In [globalTeardown](https://github.com/wix/yoshi/blob/master/packages/jest-environment-yoshi-puppeteer/globalTeardown.js#L13) we kill spawned server - this will not kill its child processes, causeing some tests to hang on travis-ci...

This PR tries to solve the problem using what was suggested [here](https://azimi.me/2014/12/31/kill-child_process-node-js.html)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ branches:
   only:
     - master
     - /^version_\d+\.x$/
+    - 796-j
```
Comment (netanelgilad) [inline .travis.yml]: need to remove this before merge.

**Comment by yanivefraim** [comment]
plz do not merge, yet... I'm trying to refactor something...

**Comment by yanivefraim** [comment]
I ended up using `killSpawnProcessAndHisChildren` for killing child process and its children

File: packages/jest-environment-yoshi-puppeteer/globalTeardown.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ const fs = require('fs-extra');
 const { WS_ENDPOINT_PATH } = require('./constants');
 const { shouldRunE2Es } = require('./utils');
 const cdnProxy = require('./cdnProxy');
+const { killSpawnProcessAndHisChildren } = require('yoshi-helpers');
```
Comment (ronami) [inline packages/jest-environment-yoshi-puppeteer/globalTeardown.js]: how about removing `killSpawnProcessAndHisChildren ` and using https://github.com/dwyl/terminate instead?

Comment (yanivefraim) [reply packages/jest-environment-yoshi-puppeteer/globalTeardown.js]: done...

**Pull request merged**